engine/classes/Elgg/Database/MetadataTable.php

```diff
@@ -391,7 +391,23 @@ public function getRowsForGuids(array $guids) {
 	 * @return bool|null true on success, false on failure, null if no metadata to delete.
 	 */
 	public function deleteAll(array $options) {
-		if (!_elgg_is_valid_options_for_batch_operation($options, 'metadata')) {
+		$required = [
+			'guid', 'guids',
+			'metadata_name', 'metadata_names',
+			'metadata_value', 'metadata_values'
+		];
+		
+		$found = false;
+		foreach ($required as $key) {
+			// check that it exists and is something.
+			if (isset($options[$key]) && $options[$key]) {
+				$found = true;
+				break;
+			}
+		}
+		
+		if (!$found) {
+			// requirements not met
```

Comment:
why isn't this in a protected function (just like in AnnotationsTable) that way it's easier to test ;)

Reply:
and ``elgg_is_empty()``

Reply:
code is just used once.. so not separating the function... elgg_is_empty() added to all the places